pkg/tfbridge/README.md

```diff
@@ -0,0 +1,148 @@
+# tfbridge
+
+## Automatic Aliasing (`ApplyAutoAliases`)
+
+Automatically applies backwards compatibility best practices.
+
+The goal is to prevent breaking changes from Pulumi maintainers or from the upstream
+provider from causing breaking changes in minor version bumps. We do this by deferring
+certain types of breaking changes to major versions.
+
+ApplyAutoAliases attempts to mitigate 3 types of unwanted breaking changes:
+
+- The token mapping for a resource has changed. For example:
+
+	The maintainer is correcting a typo in a manual resource mapping.
+
+- The token mapping for a resource has changed, and a major update caused us to remove
+the old name from the schema.
+
+- The upstream provider has added or removed `MaxItems: 1` from a field.
+
+`ApplyAutoAliases` applies three mitigation strategies: one for each breaking change it
+is attempting to mitigate.
+
+- Call `ProviderInfo.RenameResourceWithAlias` or `ProviderInfo.RenameDataSource`: This
+creates a "hard alias", preventing the user from experiencing a breaking change between
+major versions.
```

Comment:
Should say "minor versions", or needs more words to describe that we use hard aliases for all the minor and patch versions that happen, e.g after v1.x.x and before v2.x.x.